File: src/posix-extras-dbus/libqdbusunixfiledescriptor.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html
pub const qdbusunixfiledescriptor = struct {
    /// New constructs a new QDBusUnixFileDescriptor object.
    ///
    ///
    pub fn New() QtC.QDBusUnixFileDescriptor {
        return qtc.QDBusUnixFileDescriptor_new();
    }

    /// New2 constructs a new QDBusUnixFileDescriptor object.
    ///
    /// ``` fileDescriptor: i32 ```
    pub fn New2(fileDescriptor: i32) QtC.QDBusUnixFileDescriptor {
        return qtc.QDBusUnixFileDescriptor_new2(@intCast(fileDescriptor));
    }

    /// New3 constructs a new QDBusUnixFileDescriptor object.
    ///
    /// ``` other: QtC.QDBusUnixFileDescriptor ```
    pub fn New3(other: ?*anyopaque) QtC.QDBusUnixFileDescriptor {
        return qtc.QDBusUnixFileDescriptor_new3(@ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html#operator-eq)
    ///
    /// ``` self: QtC.QDBusUnixFileDescriptor, other: QtC.QDBusUnixFileDescriptor ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QDBusUnixFileDescriptor_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html#swap)
    ///
    /// ``` self: QtC.QDBusUnixFileDescriptor, other: QtC.QDBusUnixFileDescriptor ```
    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QDBusUnixFileDescriptor_Swap(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html#isValid)
    ///
    /// ``` self: QtC.QDBusUnixFileDescriptor ```
    pub fn IsValid(self: ?*anyopaque) bool {
        return qtc.QDBusUnixFileDescriptor_IsValid(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html#fileDescriptor)
    ///
    /// ``` self: QtC.QDBusUnixFileDescriptor ```
    pub fn FileDescriptor(self: ?*anyopaque) i32 {
        return qtc.QDBusUnixFileDescriptor_FileDescriptor(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html#setFileDescriptor)
    ///
    /// ``` self: QtC.QDBusUnixFileDescriptor, fileDescriptor: i32 ```
    pub fn SetFileDescriptor(self: ?*anyopaque, fileDescriptor: i32) void {
        qtc.QDBusUnixFileDescriptor_SetFileDescriptor(@ptrCast(self), @intCast(fileDescriptor));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html#giveFileDescriptor)
    ///
    /// ``` self: QtC.QDBusUnixFileDescriptor, fileDescriptor: i32 ```
    pub fn GiveFileDescriptor(self: ?*anyopaque, fileDescriptor: i32) void {
        qtc.QDBusUnixFileDescriptor_GiveFileDescriptor(@ptrCast(self), @intCast(fileDescriptor));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html#takeFileDescriptor)
    ///
    /// ``` self: QtC.QDBusUnixFileDescriptor ```
    pub fn TakeFileDescriptor(self: ?*anyopaque) i32 {
        return qtc.QDBusUnixFileDescriptor_TakeFileDescriptor(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html#isSupported)
    ///
    ///
    pub fn IsSupported() bool {
        return qtc.QDBusUnixFileDescriptor_IsSupported();
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdbusunixfiledescriptor.html#dtor.QDBusUnixFileDescriptor)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QDBusUnixFileDescriptor ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QDBusUnixFileDescriptor_Delete(@ptrCast(self));
    }
};

File: src_original/posix-extras-dbus/libqdbusunixfiledescriptor.cpp
#include <QDBusUnixFileDescriptor>
#include <qdbusunixfiledescriptor.h>
#include "libqdbusunixfiledescriptor.h"
#include "libqdbusunixfiledescriptor.hxx"

QDBusUnixFileDescriptor* QDBusUnixFileDescriptor_new() {
    return new QDBusUnixFileDescriptor();
}

QDBusUnixFileDescriptor* QDBusUnixFileDescriptor_new2(int fileDescriptor) {
    return new QDBusUnixFileDescriptor(static_cast<int>(fileDescriptor));
}

QDBusUnixFileDescriptor* QDBusUnixFileDescriptor_new3(const QDBusUnixFileDescriptor* other) {
    return new QDBusUnixFileDescriptor(*other);
}

void QDBusUnixFileDescriptor_OperatorAssign(QDBusUnixFileDescriptor* self, const QDBusUnixFileDescriptor* other) {
    self->operator=(*other);
}

void QDBusUnixFileDescriptor_Swap(QDBusUnixFileDescriptor* self, QDBusUnixFileDescriptor* other) {
    self->swap(*other);
}

bool QDBusUnixFileDescriptor_IsValid(const QDBusUnixFileDescriptor* self) {
    return self->isValid();
}

int QDBusUnixFileDescriptor_FileDescriptor(const QDBusUnixFileDescriptor* self) {
    return self->fileDescriptor();
}

void QDBusUnixFileDescriptor_SetFileDescriptor(QDBusUnixFileDescriptor* self, int fileDescriptor) {
    self->setFileDescriptor(static_cast<int>(fileDescriptor));
}

void QDBusUnixFileDescriptor_GiveFileDescriptor(QDBusUnixFileDescriptor* self, int fileDescriptor) {
    self->giveFileDescriptor(static_cast<int>(fileDescriptor));
}

int QDBusUnixFileDescriptor_TakeFileDescriptor(QDBusUnixFileDescriptor* self) {
    return self->takeFileDescriptor();
}

bool QDBusUnixFileDescriptor_IsSupported() {
    return QDBusUnixFileDescriptor::isSupported();
}

void QDBusUnixFileDescriptor_Delete(QDBusUnixFileDescriptor* self) {
    delete self;
}
